### PR TITLE
Add rangetool

### PR DESCRIPTION
**Pull request by Ashton-Sidhu**
Hi @PatrikHlobil !

I'm proposing inclusion of this new rangetool feature, that allows you to scroll along the x axis of your graph providing a better user experience for graphs with a large x axis. 

Right now I have it enabled for line plots ( but is theoretically implemented for anything that uses the _base_lineplot) as that's where it's most common and makes the most sense (maybe bar charts as well? ), but can easily be extended out and generalized to all graphs.

Let me know what you think!

### CODE REVIEW

**Comment by PatrikHlobil** [comment]
Wow, really cool feature. Big thanks for contributing again to this projects ;).

Didn't even know that Bokeh can do that out of the box now. Hadn't much time to play around with it in the last time. I am working right now at a feature where you can pass a Bokeh figure as parameter and to plot_bokeh, such that you can plot several glyphs on top of each other. Let's see how this works out, I guess I have to add some attribute to the figure such that I can see which is a real figure and which is the rangetool figure.

I will try to review it right now and give you feedback as soon as I'm done, a quick check from my side looks good ;)..

Best Patrik

File: README.md [hunk 1]
```diff
@@ -253,6 +254,17 @@ df.plot_bokeh.line(
 
 ![ApplevsGoogle_3](Documentation/Images/ApplevsGoogle_3.png)
 
+#### Lineplot with rangetool
+
+```python
+df = pd.DataFrame(np.random.randn(1000, 4), index=ts.index, columns=list('ABCD'))
```
Comment (PatrikHlobil) [inline README.md]: ts.index --> ts was never defined !

File: README.md [hunk 2]
```diff
@@ -253,6 +254,17 @@ df.plot_bokeh.line(
...
+df = df.cumsum()
+
+df.plot_bokeh(rangetool=True)
+```
+
+![rangetool](Documentation/Images/rangetool.png)
```
Comment (PatrikHlobil) [inline README.md]: Would be nice if you could add a rangetool.gif animation, such that everyone can see how cool it works ;)

File: pandas_bokeh/plot.py [hunk 3]
```diff
@@ -928,18 +941,42 @@ def _base_lineplot(
...
-        marker = kwargs["marker"]
-        del kwargs["marker"]
-    else:
-        marker = "circle"
-
-    # Add line (and optional scatter glyphs) to figure:
+    select = None
+     # Add line (and optional scatter glyphs) to figure:
     linetype = getattr(p, linetype.lower())
+    marker = kwargs.pop("marker", "circle")
```
Comment (PatrikHlobil) [inline pandas_bokeh/plot.py]: very nice refactoring ;)

File: pandas_bokeh/plot.py [hunk 4]
```diff
@@ -206,6 +210,8 @@ def plot(
             **kwargs
         )
 
+    if rangetool:
+        x_axis_location = "above"
```
[the remark this answers is not in the document]
Comment (Ashton-Sidhu) [reply pandas_bokeh/plot.py]: I definitely agree with those 2 plots and in the future I would like to add this feature to at least bar plots for sure. In the meantime I did refactor the code into a function for future use!

File: pandas_bokeh/plot.py [hunk 5]
```diff
@@ -461,7 +468,7 @@ def plot(
 
     # Add Glyphs to Plot:
     if kind == "line":
-        p = lineplot(
+        p, select = lineplot(
```
[the remark this answers is not in the document]
Comment (Ashton-Sidhu) [reply pandas_bokeh/plot.py]: Done!

File: pandas_bokeh/plot.py [hunk 6]
```diff
@@ -905,7 +914,11 @@ def plot(
 
     # Display plot if wanted
     if show_figure:
-        show(p)
+         # Display range tool if wanted, otherwise display without
+        if rangetool and kind == 'line':
```
[the remark this answers is not in the document]
Comment (Ashton-Sidhu) [reply pandas_bokeh/plot.py]: Fixed! Thanks for the tip, this solution is much more robust.

File: README.md (hunk 1 above)
Comment (Ashton-Sidhu) [reply README.md]: Fixed, thanks for catching!

File: README.md (hunk 2 above)
Comment (Ashton-Sidhu) [reply README.md]: Done!

**Comment by PatrikHlobil** [comment]
Hi @Ashton-Sidhu ,

Awesome and thanks again for your contribution 😉. I hope that I can release version 0.4 soon.

I will merge now on master.

Best Patrik

**Pull request merged**